tests/parser/functions/test_slice.py

```diff
@@ -146,15 +152,19 @@ def _get_contract():
     if (
         (start + length > data_length and literal_start and literal_length)
         or (literal_length and length > data_length)
-        or (location == "literal" and len(bytesdata) > length_bound)
+        or (
+            location == "literal"
+            and len(bytesdata) > length_bound
+            and ((literal_length and length > length_bound) or (not literal_length))
```

Comment:
can this just be
```suggestion
            and (literal_length and length > length_bound)
```

Reply:
No, this is a falsifying example (which I originally thought was a bug):
```
@external
def do_slice(inp: Bytes[1], start: uint256, length: uint256) -> Bytes[1]:
    return slice(b'\x00\x00', 0, length)
```